linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -673,19 +678,43 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
   private def transformClosureCommon(arrow: Boolean,
       captureParams: List[ParamDef], params: List[ParamDef],
       restParam: Option[ParamDef], body: Tree,
-      newCaptureValues: List[Tree])(
-      implicit scope: Scope, pos: Position): Closure = {
+      tcaptureValues: List[PreTransform])(cont: PreTransCont)(
+      implicit scope: Scope, pos: Position): TailRec[Tree] = {
 
     val thisType = if (arrow) NoType else AnyType
-    val (allNewParams, newBody) = transformIsolatedBody(None, thisType,
-        captureParams ++ params ++ restParam, AnyType, body, scope.implsBeingInlined)
-    val (newCaptureParams, newParams, newRestParam) = {
-      val (c, t) = allNewParams.splitAt(captureParams.size)
-      if (restParam.isDefined) (c, t.init, Some(t.last))
-      else (c, t, None)
+
+    val (allNewParamsWithUse, newBody) =
+      transformIsolatedBody(None, thisType, captureParams ++ params ++ restParam, AnyType, body,
+        scope.implsBeingInlined)
+
+    val (newCaptureParamsWithUse, newParamsWithUse) = allNewParamsWithUse.splitAt(captureParams.size)
+
+    val (newParams, newRestParam) = {
+      val t = newParamsWithUse.map(_._1).toList
+      if (restParam.isDefined) (t.init, Some(t.last))
+      else (t, None)
     }
 
-    Closure(arrow, newCaptureParams, newParams, newRestParam, newBody, newCaptureValues)
+    val captureBindings =
```

Comment:
```suggestion
    val captureBindings = {
```
(with the corresponding closing `}`)

As per our coding style guide, any indented multiline expression must be surrounded by braces.